Software/app/basic_azurertos/main.c

```diff
@@ -0,0 +1,190 @@
+ * limitations under the License.
+ */
+
+/**
+ * @file main.c
+ *
+ * @copyright Copyright (c) 2021 The Things Industries B.V.
+ *
+ */
+
+#include "tx_api.h"
+#include "app.h"
+
+static void SystemClock_Config(void);
+static void Error_Handler(void);
+
+static int32_t Queue_value = 100;
+
+/* Define the ThreadX object control blocks */
+TX_THREAD thread_1;
+TX_THREAD thread_2;
+TX_QUEUE queue_0;
+TX_BYTE_POOL byte_pool_0;
+
+/* Define thread prototypes */
+void thread_1_entry(ULONG thread_input);
+void thread_2_entry(ULONG thread_input);
+
+int main(int argc, char **argv)
+{
+    /* Reset of all peripherals, Initializes the Flash interface and the Systick */
+    HAL_Init();
+
+    /* Configure the system clock */
+    SystemClock_Config();
+
+#if (GNSE_TINY_TRACER_ENABLE)
+    /* Initialize Tracer/Logger */
+    GNSE_TRACER_INIT()
+#endif
+
+    /* Initialize LED */
+    GNSE_BSP_LED_Init(LED_BLUE);
+
+    /* Enter the ThreadX kernel  */
+    tx_kernel_enter();
+}
+
+/* Define what the initial system looks like  */
+void tx_application_define(void *first_unused_memory)
+{
+
+    CHAR *pointer;
+
+    /* Create a byte memory pool from which to allocate the thread stacks  */
+    tx_byte_pool_create(&byte_pool_0, "byte pool 0", first_unused_memory,
+                        APP_DEMO_BYTE_POOL_SIZE);
+
+    /* Allocate the stack for thread 1  */
+    tx_byte_allocate(&byte_pool_0, (VOID **)&pointer, APP_DEMO_STACK_SIZE,
+                     TX_NO_WAIT);
+
+    /* Create threads 1 and 2. These threads pass information through a ThreadX
+     message queue */
+    tx_thread_create(&thread_1, "thread 1", thread_1_entry, 1, pointer,
+                     APP_DEMO_STACK_SIZE, 2, 2, TX_NO_TIME_SLICE, TX_AUTO_START);
+
+    /* Allocate the stack for thread 2 */
+    tx_byte_allocate(&byte_pool_0, (VOID **)&pointer, APP_DEMO_STACK_SIZE,
+                     TX_NO_WAIT);
+
+    tx_thread_create(&thread_2, "thread 2", thread_2_entry, 2, pointer,
+                     APP_DEMO_STACK_SIZE, 1, 1, TX_NO_TIME_SLICE, TX_AUTO_START);
+
+    /* Allocate the message queue  */
+    tx_byte_allocate(&byte_pool_0, (VOID **)&pointer,
+                     APP_DEMO_QUEUE_SIZE * sizeof(ULONG), TX_NO_WAIT);
+
+    /* Create the message queue shared by threads 1 and 2  */
+    tx_queue_create(&queue_0, "queue 0", TX_1_ULONG, pointer,
+                    APP_DEMO_QUEUE_SIZE * sizeof(ULONG));
+}
+
+void thread_1_entry(ULONG thread_input)
+{
+    UINT status;
+
+    /* This thread simply sends messages to a queue shared by thread 2  */
+    while (1)
+    {
+        /* Send message to queue 0 */
+        APP_PPRINTF(("\r\n Send Tx msg \r\n"));
+        status = tx_queue_send(&queue_0, &Queue_value,
+                               TX_WAIT_FOREVER);
+
+        /* Check completion status */
+        if (status == TX_SUCCESS)
+        {
+            tx_thread_sleep(APP_TX_DELAY);
+        }
+        else
+        {
+            break;
```

Comment:
Is it wise to break out of the while loop if the queue sending fails? What will happen when a user accidentally configures it wrong?

Reply:
Yes this is intentional, because if there is bad configuration, the app should not function